src/SAML2/XML/AbstractXMLElement.php

```diff
@@ -101,6 +105,138 @@ public function isEmptyElement(): bool
+    }
+
+
+    /**
+     * @param DOMElement  $xml The element where we should search for the attribute.
+     * @param string      $name The name of the attribute.
+     * @param string|null $default The default to return in case the attribute does not exist and it is optional.
+     *
+     * @return bool|null
+     * @throws \InvalidArgumentException
+     */
+    public static function getBooleanAttribute(DOMElement $xml, string $name, ?string $default = ''): ?bool
+    {
+        $value = self::getAttribute($xml, $name, $default);
+        if ($value === null) {
+            return null;
+        }
+        Assert::oneOf(
+            $value,
+            ['0', '1', 'false', 'true'],
+            'The \'' . $name . '\' attribute of ' . static::NS_PREFIX . ':' . self::getClassName(static::class) .
+            ' must be boolean.'
+        );
+        return in_array($value, ['1', 'true'], true);
+    }
+
+
+    /**
+     * Get the integer value of an attribute from a given element.
+     *
+     * @param DOMElement  $xml The element where we should search for the attribute.
+     * @param string      $name The name of the attribute.
+     * @param string|null $default The default to return in case the attribute does not exist and it is optional.
+     *
+     * @return int|null
+     */
+    public static function getIntegerAttribute(DOMElement $xml, string $name, ?string $default = ''): ?int
+    {
+        $value = self::getAttribute($xml, $name, $default);
+        if ($value === null) {
+            return null;
+        }
+        Assert::numeric(
+            $value,
+            'The \'' . $name . '\' attribute of ' . static::NS_PREFIX . ':' . self::getClassName(static::class) .
+            ' must be numerical.'
+        );
+        return intval($value);
+    }
+
+
+    /**
+     * Static method that processes a fully namespaced class name and returns the name of the class from it.
+     *
+     * @param string $class
+     *
+     * @return string
+     */
+    protected static function getClassName(string $class): string
+    {
+        return join('', array_slice(explode('\\', $class), -1));
+    }
+
+
+    /**
+     * Get the XML local name of the element represented by this class.
+     *
+     * @return string
+     */
+    public function getLocalName(): string
+    {
+        return self::getClassName(get_class($this));
+    }
+
+
+    /**
+     * Get the XML qualified name (prefix:name) of the element represented by this class.
+     *
+     * @return string
+     */
+    public function getQualifiedName(): string
+    {
+        return static::NS_PREFIX . ':' . $this->getLocalName();
+    }
+
+
+    /**
+     * Extract localized names from a the children of a given element.
+     *
+     * @param DOMElement $parent The element we want to search.
+     *
+     * @return array An array of objects of class $class.
+     */
+    public static function extractFromChildren(\DOMElement $parent): array
```

Comment:
I'm still in doubt about this method's name. What do you think about something like `getChildrenOfType()`?

Reply:
I would keep it as is, because 'Type' may raise confusion with SAML types.. I do see the phpdoc for this method needs a polish!

Reply:
What about `getChildrenOfClass()`?

Reply:
Renamed & touched up the method